materialize-snowflake/snowflake.go

```diff
@@ -553,6 +564,113 @@ func (d *transactor) Acknowledge(ctx context.Context) (*pf.ConnectorState, error
 		d.deleteFiles(ctx, []string{item.StagedDir})
 	}
 
+	// Keep asking for a report on the files that have been submitted for processing
+	// until they have all been successful, or an error has been thrown
+	var tries = 0
+	var maxTries = 10
+	for len(pipes) > 0 {
+		for pipeName, record := range pipes {
+			report, err := d.pipeClient.InsertReport(pipeName, d.pipeMarkers[pipeName])
+			if err != nil {
+				return nil, fmt.Errorf("snowpipe: insertReports: %w", err)
+			}
+
+			// If the files have already been loaded, when we submit a request to
+			// load those files again, our request will not show up in reports.
+			// One way to find out whether the files were successfully loaded is to check
+			// the COPY_HISTORY to make sure they are there. If they are not, then something
+			// may be wrong.
+			if len(report.Files) == 0 {
+				if tries < maxTries {
+					tries = tries + 1
+					continue
+				}
+
+				log.WithField("tries", tries).Info("snowpipe: no files in report, fetching copy history from warehouse")
+
+				var fileNames = make([]string, len(record.files))
+				for i, f := range record.files {
+					fileNames[i] = f.Path
+				}
+
+				query, err := RenderCopyHistoryTemplate(record.binding.target, fileNames, record.startTime, d.templates.copyHistory)
+				if err != nil {
+					return nil, fmt.Errorf("snowpipe: rendering copy history: %w", err)
+				}
+
+				rows, err := d.store.conn.QueryContext(ctx, query)
+				if err != nil {
+					return nil, fmt.Errorf("snowpipe: fetching copy history: %w", err)
+				}
+				defer rows.Close()
+
+				var (
+					fileName          string
+					status            string
+					firstErrorMessage *string
+				)
+				for rows.Next() {
+					if err := rows.Scan(&fileName, &status, &firstErrorMessage); err != nil {
+						return nil, fmt.Errorf("scanning copy history row: %w", err)
+					}
+
+					log.WithFields(log.Fields{
+						"fileName":          fileName,
+						"status":            status,
+						"firstErrorMessage": firstErrorMessage,
+						"pipeName":          pipeName,
+					}).Info("snowpipe: copy history row")
```

Comment:
Since this is not supposed to happen frequently, I think it's fine to leave this as info so we can troubleshoot when this happens